### ISSUE TITLE
Add ability to perform a batch build from the command line.

### ISSUE THREAD

**Issue by mrudat** [ousnius/BodySlide-and-Outfit-Studio#112]
It would be nice if we could have sufficient options added to allow performing a batch build from the command line; it would handily work around the fact that listing files/directories in Mod Organiser 2 is currently broken, and would make it possible to create a plugin for Mod Organiser that could rebuild target nifs, should either the sliderpreset or sliderdata have changed (eg, installing an updated outfit).

I believe that we would need to have something like:

``` c++
/* bodySlideApp.h */
    bool straightOutfitStudio = false;
    bool straightBuild = false;
    wxString buildTargetDirectory = null;
    wxString buildPresetName = null;
    wxString buildOutfitName = null;
    wxString buildOutfitGroupName = null;

    /* Data Managers */
    SliderManager sliderManager;
/* --8<-- */
static const wxCmdLineEntryDesc g_cmdLineDesc[] = {
    { wxCMD_LINE_SWITCH, "os", "outfitstudio", "launches straight into outfit studio" },
    { wxCMD_LINE_SWITCH, "b", "build", "Build the specified outfits" },
    { wxCMD_LINE_OPTION, "td", "targetDirectory", "Where to build the outfits, defaults to the game data directory", wxCMD_LINE_VAL_STRING },
    { wxCMD_LINE_OPTION, "pn", "presetName", "The name of the preset used to build the outfits, defaults to the last used preset", wxCMD_LINE_VAL_STRING },
    { wxCMD_LINE_OPTION, "on", "outfitName", "The name of the outfit to build, if not specified, builds the last used outfit group", wxCMD_LINE_VAL_STRING },
    { wxCMD_LINE_OPTION, "og", "outfitGroup", "The name of the outfit group to build, if not specified builds the last used outfit group", wxCMD_LINE_VAL_STRING },
    { wxCMD_LINE_NONE }
};
```

``` c++
/* BodySlideApp.cpp OnInit */
    }

    if (straightOutfitStudio)
        LaunchOutfitStudio();
    if (straightBuild)
        DoBuild();

    wxLogMessage("BodySlide initialized.");
/* --8<-- */
bool BodySlideApp::OnCmdLineParsed(wxCmdLineParser& parser) {
    straightOutfitStudio = parser.Found("os");
    straightBuild = parser.Found("b");
    parser.Found("td",buildTargetDirectory);
    parser.Found("pn",buildPresetName);
    parser.Found("on",buildOutfitName);
    parser.Found("og",buildOutfitGroupName);
    return true;
}
```

a modified chunk of  `OnBatchBuild`

``` c++
void BodySlideApp::DoBuild() {
    vector<string> toBuild;

    bool tri = false; // TODO add option for this

    // FIXME apply supplied options so that the following will work.

    app->GetFilteredOutfits(toBuild);

    map<string, string> failedOutfits;
    int ret;

    if (buildTargetDirectory != null) 
        ret = app->BuildListBodies(toBuild, failedOutfits, false, tri, buildTargetDirectory + "\\");
    else
        ret = app->BuildListBodies(toBuild, failedOutfits, false, tri);

    if (ret == 0) {
        wxLogMessage("All sets processed successfully!");
        // cout << "All sets processed successfully!" << endl;
    }
    else if (ret == 3) {
        for (auto &e : failedOutfits) {
            wxLogError("Failed to build '%s': %s", e.first, e.second);
            // printf("Failed to build '%s': %s", e.first,+ e.second);
        }
    }
}
```

**Comment by ousnius** [comment]
@mrudat:
Implemented in 17be68c.

**Example usage:**
`BodySlide.exe -gbuild "CBBE Bodies" -p "CBBE Athletic" -t "C:\gbuild\\"`
(mind the double \ at the end!)

Does this work in a way that satisfies your needs?

**Comment by mrudat** [comment]
Yes, that's pretty much what I was looking for, thanks. =) Now, to start writing a plugin for Mod Organiser...
